stock_picking_group_by_date_deadline/models/stock_move.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023 Foodles (http://www.foodles.co).
+# @author Pierre Verkest <pierreverkest84@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from datetime import timedelta
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class DeadlineDate:
+    """purpose of this class is to wrap datetime
+    in the id attribute as keys are expected in
+    id attribute
+    """
+
+    id = False
+
+    def __init__(self, dt):
+        self.id = dt
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    date_deadline = fields.Datetime(inverse="_inverse_date_deadline")
```

Comment:
Why not create a dedicated field for this need ?

Reply:
because Odoo use this date_deadline field to distinguish move to merge or not https://github.com/odoo/odoo/blob/14.0/addons/stock/models/stock_move.py#L795-L800 so I wanted to change that value to gives more chance to merge them together as I "round" the value.

I've the feeling this is easier to maintain and configure changing the deadline date than change the behaviour to define a range of time to  allow to merge/group or not. The other possible things with this implementation is to overwrite how to compute deadline date to add more business logic (likes exclude closed days or days of to move the day before)